operations/secure_random/secure_random.cc

```diff
@@ -0,0 +1,203 @@
+#include "tensorflow/core/framework/op.h"
+#include "tensorflow/core/kernels/bounds_check.h"
+#include "tensorflow/core/framework/shape_inference.h"
+#include "tensorflow/core/framework/op_kernel.h"
+#include "tensorflow/core/lib/random/random_distributions.h"
+#include "sodium.h"
+
+using namespace tensorflow;
+
+using shape_inference::DimensionHandle;
+using shape_inference::ShapeHandle;
+
+#define CHACHABLOCKSIZE 64
+
+static Status SecureRandomShape(shape_inference::InferenceContext* context) {
+  // Check seed shape
+  ShapeHandle seed;
+  TF_RETURN_IF_ERROR(context->WithRank(context->input(1), 1, &seed));
+  DimensionHandle unused;
+  TF_RETURN_IF_ERROR(context->WithValue(context->Dim(seed, 0), 8, &unused));
+
+  // Set output shape
+  ShapeHandle out;
+  TF_RETURN_IF_ERROR(context->MakeShapeFromShapeTensor(0, &out));
+  context->set_output(0, out);
+  return Status::OK();
+}
+
+REGISTER_OP("SecureRandom")
+    .Input("shape: T")
+    .Input("seed: Tseed")
+    .Input("minval: dtype")
```

Comment:
why `dtype` btw?

Reply:
The syntax is `<tensor name>: <tensor type>` and the `<tensor type>` can be references from the attributes below. Thought I'd just call that attribute `dtype` because 3 of the inputs rely on it.